rust/main/hyperlane-base/src/metrics/core.rs

```diff
@@ -112,6 +113,16 @@ impl CoreMetrics {
             registry
         )?;
 
+        let latest_tree_insertion_count = register_int_gauge_vec_with_registry!(
+            opts!(
+                namespaced!("latest_tree_insertion_count"),
```

Comment:
as mentioned irl, could see this being interpreted as "how many leaves were inserted in the tree since the relayer started up" so would prefer something like `max_merkle_tree_index` or something

Reply:
ya, renamed it to `latest_leaf_index` which is closer to the associated gauge `max_leaf_index_gauge` in MerkleTreeProcessorMetrics here https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/5074/commits/ef8afa0beccbc96426506752a9616b1d00be72d1